turbomoleio/core/molecule.py

```diff
@@ -69,7 +88,71 @@ def get_mol_and_indices_frozen(string):
         if len(s) == 5 and s[-1].lower() == "f":
             frozen_indices.add(iatom)
 
-    mol = Molecule(species, coords)
+    coords = np.array(coords)
+    if cell_string is not None:
+        if lattice_string is not None:
+            raise ValueError('Only cell_string or lattice_string should be set.')
+        periodicity = int(periodic_string.strip())
+        cell = [float(sp) for sp in cell_string.strip().split()]
+        if periodicity == 1:
+            if len(cell) != 1:
+                raise ValueError('The $cell data group should contain 1 number for 1D periodic systems')
+            yrange = np.max(coords[:, 1])-np.min(coords[:, 1])
+            zrange = np.max(coords[:, 2])-np.min(coords[:, 2])
+            lattice = Lattice.orthorhombic(a=bohr_to_ang*cell[0],
+                                           b=bohr_to_ang*yrange+periodic_extension,
+                                           c=bohr_to_ang*zrange+periodic_extension)
+        elif periodicity == 2:
+            if len(cell) != 3:
+                raise ValueError('The $cell data group should contain 3 numbers for 2D periodic systems')
+            zrange = np.max(coords[:, 2])-np.min(coords[:, 2])
+            lattice = Lattice.from_parameters(a=bohr_to_ang*cell[0],
+                                              b=bohr_to_ang*cell[1],
+                                              c=bohr_to_ang*zrange+periodic_extension,
+                                              alpha=90.0, beta=90.0, gamma=cell[2])
+        elif periodicity == 3:
+            if len(cell) != 6:
+                raise ValueError('The $cell data group should contain 6 numbers for 3D periodic systems')
+            lattice = Lattice.from_parameters(a=bohr_to_ang*cell[0],
+                                              b=bohr_to_ang*cell[1],
+                                              c=bohr_to_ang*cell[2],
+                                              alpha=cell[3],
+                                              beta=cell[4],
+                                              gamma=cell[5])
+        else:
+            raise RuntimeError('Periodicity should be one of 1, 2 or 3.')
+        mol = Structure(lattice=lattice, species=species, coords=coords, coords_are_cartesian=True)
+    elif lattice_string is not None:
+        if cell_string is not None:
+            raise ValueError('Only cell_string or lattice_string should be set.')
+        lattice_string = lattice_string.strip()
+        periodicity = int(periodic_string.strip())
```

Comment:
Same as above, in principle, it will not be None for the call in PeriodicSystem.from_string and there is a check in that method. Do you think we should add another check here ?